papaya-core/src/main/java/pl/edu/agh/papaya/service/availability/AvailabilityCreationWizard.java

```diff
@@ -0,0 +1,65 @@
+package pl.edu.agh.papaya.service.availability;
+
+import java.time.Duration;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import pl.edu.agh.papaya.model.Availability;
+import pl.edu.agh.papaya.model.Sprint;
+import pl.edu.agh.papaya.model.UserInProject;
+
+@RequiredArgsConstructor
+@SuppressWarnings({"checkstyle:HiddenField", "PMD.BeanMembersShouldSerialize"})
+public class AvailabilityCreationWizard {
+
+    private final AvailabilityService availabilityService;
+
+    private Sprint sprint;
+    private UserInProject userInProject;
+    private Duration timeAvailable;
+    private Duration pastSprintRemainingTime;
+    private String notes;
+
+    public AvailabilityCreationWizard withSprint(@NonNull Sprint sprint) {
+        this.sprint = sprint;
+        return this;
+    }
+
+    public AvailabilityCreationWizard withUserInProject(@NonNull UserInProject userInProject) {
+        this.userInProject = userInProject;
+        return this;
+    }
+
+    public AvailabilityCreationWizard withTimeAvailable(@NonNull Duration timeAvailable) {
+        this.timeAvailable = timeAvailable;
+        return this;
+    }
+
+    public AvailabilityCreationWizard withPastSprintRemainingTime(@NonNull Duration pastSprintRemainingTime) {
+        this.pastSprintRemainingTime = pastSprintRemainingTime;
+        return this;
+    }
+
+    public AvailabilityCreationWizard withNotes(String notes) {
+        this.notes = notes;
+        return this;
+    }
+
+    public Availability create() {
+        Availability availability = new Availability();
+        availability.setSprint(require("sprint", sprint));
+        availability.setUserInProject(require("userInProject", userInProject));
+        availability.setTimeAvailable(require("timeAvailable", timeAvailable));
+        availability.setPastSprintRemainingTime(require("pastSprintRemainingTime", pastSprintRemainingTime));
+        availability.setNotes(notes);
+
+        availabilityService.save(availability);
+        return availability;
+    }
+
+    private <T> T require(String fieldName, T value) {
```

Comment:
To jest wyjęte do utilsów w jednym PR @Sopel97 - trzeba będzie pilnować, co się pierwsze zmerge'uje